php/libraries/OutputWrapper.class.inc

```diff
@@ -0,0 +1,185 @@
+<?php declare(strict_types=1);
+/**
+ * This class is designed to assist in writing command-line interface tools
+ * for LORIS.
+ *
+ * It contains helper functions to format output, etc.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class OutputWrapper
+{
+    // Colours associated with the bash shell.
+    // @see https://misc.flogisoft.com/bash/tip_colors_and_formatting
+    const COLOURS = [
+        'red'    => '31',
+        'yellow' => '33',
+        'green'  => '32',
+    ];
+    /**
+     * Whether this script should log its output to a file or just print to
+     * STDOUT.
+     *
+     * @var bool
+     */
+    private $_loggingEnabled = false;
+    /**
+     * The directory that will contain log files written by this script.
+     *
+     * @var SPLFileInfo
+     */
+    private $_logDirectory;
+
+    /**
+     * Create a new command line helper.
+     *
+     * @param ?SPLFileInfo $logDirectory The directory that will contain log
+     *                                   files written by this script.
+     *
+     * @return void
+     */
+    public function __construct(SPLFileInfo $logDirectory = null)
+    {
+        $this->_logDirectory = new SPLFileInfo(
+            $logDirectory ?? __DIR__ . '/tools/'
+        );
+    }
+
+    /**
+     * Print a formatted and indented bullet point.
+     *
+     * @param string $line The output.
+     *
+     * @return void
+     */
+    public function printBulletPoint(string $line): void
+    {
+        echo "\t* $line\n";
+    }
+
+    /**
+     * Prints a line of text with a newline before. Can optionally be printed
+     * to the command-line in coloured text. Valid options for colours are given
+     * in the COLOURS class constant.
+     * If logging is enabled, the line will also be written to a log file. Log
+     * files will not contain coloured text.
+     *
+     * @param string   $line   To be written.
+     * @param string   $colour A valid colour, given by COLOURS, for the output.
+     * @param resource $handle A file resource where text should be written.
+     *
+     * @return void
+     */
+    public function printLine(
+        string $line,
+        string $colour = '',
+        $handle = STDOUT
+    ) {
+        $output = $line;
+        // If a valid colour is supplied and the output is going to a TTY,
+        // colourize the output.
+        if (!empty($colour) && stream_isatty($handle)) {
```

Comment:
You want to validate `$colour` regardless of if the `$handle` is a tty, you just don't want to modify the output if it's a TTY.

Reply:
`$colour` should never be considered if the handle isn't a TTY though. We don't need it.

Reply:
But the caller still passed an invalid value to the function and violated the function's precondition.

Reply:
though call here but I think I'm slightly with @driusan 

It's just as confusing if you have a script that works fine for years when stream is not a TTY and then suddently you run it from terminal and it starts failing... I feel like it's a safer bet to have the colour validated immediately, most likely the developer would catch the error as soon as the code is changed for and invalid colour instead of only the first time it is run from a TTY.


The other option in that case would be to create a printColourfulLine and have the client decide which function to use... but i feel like we lose a lot of the streamlining with that so I'm not a big fan

Reply:
@johnsaigle if you agree with the above I created a PR on ur repo to apply the changes and fix PHAN

https://github.com/johnsaigle/Loris/pull/18

Reply:
I merged your PR on my fork so this should be resolved